From 32456cdbc7f375e5048cb75ee2a992db1e436a61 Mon Sep 17 00:00:00 2001
From: Christophe Giboudeaux <christophe@krop.fr>
Date: Sun, 20 Sep 2020 09:57:22 +0200
Subject: [PATCH] Tell the truth about private API

Mark private API with symbols only for the current patch release

This change is a port of the libqt5-qtbase patch which was
added during the Qt 5.6 cycle.

Change-Id: Iada0be6e7d4eaa3a6683b82183643c88dbfa6580
---
 cmake/QtFlagHandlingHelpers.cmake | 4 ++--
 mkspecs/features/qt_module.prf    | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/cmake/QtFlagHandlingHelpers.cmake b/cmake/QtFlagHandlingHelpers.cmake
index a71c083717..27bd349924 100644
--- a/cmake/QtFlagHandlingHelpers.cmake
+++ b/cmake/QtFlagHandlingHelpers.cmake
@@ -3,9 +3,9 @@ function(qt_internal_add_linker_version_script target)
 
     if (TEST_ld_version_script)
         if (arg_INTERNAL)
-            set(contents "Qt_${PROJECT_VERSION_MAJOR}_PRIVATE_API { *; };")
+            set(contents "Qt_${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}_PRIVATE_API { *; };")
         else()
-            set(contents "Qt_${PROJECT_VERSION_MAJOR}_PRIVATE_API {\n    qt_private_api_tag*;\n")
+            set(contents "Qt_${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}_PRIVATE_API {\n    qt_private_api_tag*;\n")
             foreach(ph ${arg_PRIVATE_HEADERS})
                 string(APPEND contents "    @FILE:${ph}@\n")
             endforeach()
diff --git a/mkspecs/features/qt_module.prf b/mkspecs/features/qt_module.prf
index 828a9621b9..805d8b301a 100644
--- a/mkspecs/features/qt_module.prf
+++ b/mkspecs/features/qt_module.prf
@@ -220,9 +220,9 @@ android: CONFIG += qt_android_deps no_linker_version_script
     QMAKE_LFLAGS += $${QMAKE_LFLAGS_VERSION_SCRIPT}$$verscript
 
     internal_module {
-        verscript_content = "Qt_$${QT_MAJOR_VERSION}_PRIVATE_API { *; };"
+        verscript_content = "Qt_$${QT_MAJOR_VERSION}.$${QT_MINOR_VERSION}.$${QT_PATCH_VERSION}_PRIVATE_API { *; };"
     } else {
-        verscript_content = "Qt_$${QT_MAJOR_VERSION}_PRIVATE_API {" \
+        verscript_content = "Qt_$${QT_MAJOR_VERSION}.$${QT_MINOR_VERSION}.$${QT_PATCH_VERSION}_PRIVATE_API {" \
                             "    qt_private_api_tag*;"
 
         private_api_headers = $$SYNCQT.PRIVATE_HEADER_FILES $$SYNCQT.QPA_HEADER_FILES
-- 
2.28.0

